components/uid.rst

```diff
@@ -283,6 +312,35 @@ entity primary keys::
 
     The ULID types and generator were introduced in Symfony 5.2.
 
+When using built-in Doctrine repository methods (e.g. ``findOneBy()``), Doctrine
+knows how to convert these ULID types to build the SQL query
+(e.g. ``->findOneBy(['user' => $user->getUlid()])``). However, when using DQL
+queries or building the query yourself, you'll need to set ``ulid`` as the type
+of the ULID parameters::
+
+    // src/Repository/ProductRepository.php
+
+    // ...
+    class ProductRepository extends ServiceEntityRepository
+    {
+        // ...
+
+        public function findUserProducts(User $user): array
+        {
+            $qb = $this->createQueryBuilder('p')
+                // ...
+                // add 'ulid' as the third argument to tell Doctrine that this is an ULID
+                ->setParameter('user', $user->getUlid(), 'ulid')
```

Comment:
Doctrine recommends to use constants in the [documentation](https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/query-builder.html#binding-parameters-to-your-query), is there an equivalent for ULID?

```php

<?php

use Doctrine\DBAL\Types\Types;

// prevents attempt to load metadata for date time class, improving performance
$qb->setParameter('date', new \DateTimeImmutable(), Types::DATE_IMMUTABLE)
```

Reply:
I can't see the type in DBAL. There's `guid` but not `uuid` or `ulid`. See https://github.com/doctrine/dbal/blob/2.12.x/lib/Doctrine/DBAL/Types/Types.php

Reply:
Does `ulid` require another dependency that Doctrine ?

It looks like GUID and UUID could be used because they are synonyms:

```
/**
 * Represents a GUID/UUID datatype (both are actually synonyms) in the database.
 */
class GuidType extends StringType
{
```

Source: https://github.com/doctrine/dbal/blob/9e37d1bdc1ee589e9baab2bb16aebf6e16a18a92/lib/Doctrine/DBAL/Types/GuidType.php#L8

But I dont know for ULID and I don't want to block the merging of this PR, feel free to keep the current code.

Reply:
Yes, it's sad that we don't have an ULID type ... so let's use strings for now. Thanks!